tools/cldr-code/src/main/java/org/unicode/cldr/util/CLDRFileOverride.java

```diff
@@ -24,16 +31,20 @@ public CLDRFileOverride(CLDRFile sourceFile, Map<String, String> valueOverrides)
      * same, since it provides for the map.
      */
     public static class XMLSourceMapOverride extends XMLSource {
-        private XMLSource delegate;
+        private final XMLSource delegate;
 
-        private Map<String, String> overrides;
+        private final Map<String, String> overrides;
 
         public XMLSourceMapOverride(XMLSource source, Map<String, String> overrides) {
             overrides.keySet().stream()
                     .forEach(
                             x -> {
                                 if (source.getValueAtDPath(x) == null) {
-                                    throw new IllegalArgumentException();
```

Comment:
@macchiati @btangmu  why is this check even here? The class comment says "the (overridden) keys must already be in the source"… but, why why is it an error to override a null DPath with a non-null one from `overrides`?

Reply:
I added a comment on https://unicode-org.atlassian.net/browse/CLDR-18453 linking back here.